### PR TITLE
feat: browserslist config file support

### PR DESCRIPTION
**Pull request by ms-fadaei**
Add browserslist config file support

## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Resolves: #9337
After supporting modern browsers config in the Nuxt, there were two things that must be worked on it too:
1. Retrieve targets browsers from browserslist config file (.browserlistrc): With this RP, now the Nuxt users can configure their target browsers from a separate file. (that is used by other packages like Autoprifixer, etc too)
2. Target browsers detection on the server, which I reported in #9337 and I think @clarkdo working on it currently

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Review by clarkdo — APPROVED** [review]
Looks good. I think we can hold this pr for now until we support customized targets for SSR

**Comment by clarkdo** [comment]
> Target browsers detection on the server, which I reported in fix(babel-preset-app): respect explicit options.targets for modern preset #9337 and I think @clarkdo working on it currently

Thanks for the reminder, I’ll find some time this week to look into it.

**Comment by pi0** [comment]
Thanks for pull request @ms-fadaei <3. I think for properly solving the issue in https://github.com/nuxt/nuxt.js/pull/9337#issuecomment-862878228, we should dynamically generate [modernBrowsers](https://github.com/nuxt/nuxt.js/blob/0742f59e2478bd3998fd4d49b6200584ee5e2721/packages/utils/src/modern.js#L4) based on updated nuxt config. Relying on external config files is anti-pattern in nuxt (since we loose single source of trust) and also it needs more boilerplate to update modern target.

**Comment by pi0** [comment]
Thanks for PR @ms-fadaei <3. I think for properly solving the issue mentioned in https://github.com/nuxt/nuxt.js/pull/9337#issuecomment-862878228, we should dynamically generate [modernBrowsers](https://github.com/nuxt/nuxt.js/blob/0742f59e2478bd3998fd4d49b6200584ee5e2721/packages/utils/src/modern.js#L4) based on specified server target. Relying on external config file like `.browserlist` is an anti-pattern in nuxt (since we lose a single source of trust) and also it needs more hassle to update the modern target. Pending for @clarkdo additions.

**Comment by ms-fadaei** [comment]
Thanks for your review @pi0. I will be happy to be part of developing this issue based on the Nuxt principles, so please let me know what can I do for improving this feature?

**Comment by gkatsanos** [comment]
I run into this while looking at the documentation:

> The default targets of [@nuxt/babel-preset-app ](https://github.com/nuxt/nuxt.js/blob/dev/packages/babel-preset-app/src/index.js)are ie: '9' in the client build, and node: 'current' in the server build. 

Do I get this correctly that unless we use our own preset config, we're shipping IE9 compatible javascript?
So the `browserslist` config in `package.json` is ignored? 

Any examples on changing this to target more modern browsers?

**Comment by danielroe** [comment]
As Nuxt 2 is currently in maintenance-mode until June 30, 2024, we are not aiming to merge any more features and so, regretfully, I am closing this PR. My apologies that we weren't able to include it in v2.17 🙏

**Pull request closed without merging**
